server/player/player.go

```diff
@@ -1584,6 +1585,21 @@ func (p *Player) Position() mgl64.Vec3 {
 	return p.pos.Load().(mgl64.Vec3)
 }
 
+// Velocity returns the players current velocity. If there is an attached session, this will be empty.
+func (p *Player) Velocity() mgl64.Vec3 {
+	return p.vel.Load().(mgl64.Vec3)
+}
+
+// UpdateVelocity updates the players velocity. If there is an attached session, this will just send
+// the velocity to the player session for the player to update.
+func (p *Player) UpdateVelocity(velocity mgl64.Vec3) {
+	if p.s == session.Nop {
```

Comment:
Should use `(*player.Player).session()` here to prevent race conditions

Reply:
Fixed.